programs/openbook-v2/src/instructions/consume_events.rs

```diff
@@ -66,17 +70,24 @@ pub fn consume_events(ctx: Context<ConsumeEvents>, limit: usize) -> Result<()> {
     let mut event_queue = ctx.accounts.event_queue.load_mut()?;
     let remaining_accs = &ctx.remaining_accounts;
 
-    // Iterate over event_queue
-    for _ in 0..limit {
-        let event = match event_queue.peek_front() {
-            None => break,
+    let slots_to_consume: Vec<usize> = match slots {
+        Some(slots) => slots.into_iter().take(limit).collect(),
+        None => event_queue
+            .iter()
+            .take(limit)
+            .map(|(_event, slot)| slot)
+            .collect(),
+    };
```

Comment:
minor: there might be a way to avoid creating a `Vec<usize>` here, since the only thing we need is to iterate over the slots?

Reply:
i've tried to use an interator, but I had no luck returning the same type from the two branches of the match